collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -2109,6 +2110,7 @@ private void finishAndReturnInstance() {
     }
 
     private void exit() {
+        ReferenceManager.instance().reset();
```

Comment:
I think this should move to `FormEntryViewModel#exit`. I'm hoping in the future we're have some kind of `FormEntryDataService` than can look after loading and saving forms as well as clean up and I imagine that would get called from there.

Reply:
Done.